dev/devicelab/lib/framework/utils.dart

```diff
@@ -0,0 +1,403 @@
+  proc.stderr.listen((List<int> data) {
+    stderr.add(data);
+  });
+  String output = await UTF8.decodeStream(proc.stdout);
+  int exitCode = await proc.exitCode;
+
+  if (exitCode != 0 && !canFail)
+    fail('Executable failed with exit code $exitCode.');
+
+  return output.trimRight();
+}
+
+Future<int> flutter(String command,
+    {List<String> options: const <String>[], bool canFail: false}) {
+  List<String> args = <String>[command]..addAll(options);
+  return exec(path.join(flutterDirectory.path, 'bin', 'flutter'), args,
+      canFail: canFail);
+}
+
+String get dartBin =>
+    path.join(flutterDirectory.path, 'bin/cache/dart-sdk/bin/dart');
+
+Future<int> dart(List<String> args) => exec(dartBin, args);
+
+Future<int> pub(String command) {
+  return exec(
+      path.join(flutterDirectory.path, 'bin/cache/dart-sdk/bin/pub'),
+      <String>[command]);
+}
+
+Future<dynamic> inDirectory(dynamic directory, Future<dynamic> action()) async {
+  String previousCwd = cwd;
+  try {
+    cd(directory);
+    return await action();
+  } finally {
+    cd(previousCwd);
+  }
+}
+
+void cd(dynamic directory) {
+  Directory d;
+  if (directory is String) {
+    cwd = directory;
+    d = dir(directory);
+  } else if (directory is Directory) {
+    cwd = directory.path;
+    d = directory;
+  } else {
+    throw 'Unsupported type ${directory.runtimeType} of $directory';
+  }
+
+  if (!d.existsSync())
+    throw 'Cannot cd into directory that does not exist: $directory';
+}
+
+Directory get flutterDirectory => dir('../..').absolute;
+
+String requireEnvVar(String name) {
+  String value = Platform.environment[name];
+
+  if (value == null) fail('$name environment variable is missing. Quitting.');
+
+  return value;
+}
+
+dynamic/*=T*/ requireConfigProperty/*<T>*/(
+    Map<String, dynamic/*<T>*/ > map, String propertyName) {
+  if (!map.containsKey(propertyName))
+    fail('Configuration property not found: $propertyName');
+
+  return map[propertyName];
+}
+
+String jsonEncode(dynamic data) {
+  return new JsonEncoder.withIndent('  ').convert(data) + '\n';
+}
+
+Future<Null> getFlutter(String revision) async {
+  section('Get Flutter!');
+
+  if (exists(flutterDirectory)) {
+    rrm(flutterDirectory);
+  }
+
+  await inDirectory(flutterDirectory.parent, () async {
+    await exec('git', <String>['clone', 'https://github.com/flutter/flutter.git']);
+  });
+
+  await inDirectory(flutterDirectory, () async {
+    await exec('git', <String>['checkout', revision]);
+  });
+
+  await flutter('config', options: <String>['--no-analytics']);
+
+  section('flutter doctor');
+  await flutter('doctor');
+
+  section('flutter update-packages');
+  await flutter('update-packages');
+}
+
+void checkNotNull(Object o1,
+    [Object o2 = 1,
+    Object o3 = 1,
+    Object o4 = 1,
+    Object o5 = 1,
+    Object o6 = 1,
+    Object o7 = 1,
+    Object o8 = 1,
+    Object o9 = 1,
+    Object o10 = 1]) {
+  if (o1 == null) throw 'o1 is null';
```

Comment:
I ran `dartfmt` across this code. My impression was that we can now use it on Flutter code. There was an email from @sethladd  inviting us to try it. Is it not ready? Would be great to get rid of the time-consuming style-related review comments.